### PR TITLE
Tuple names shouldn't affect whether a type is a base of another

### PR DESCRIPTION
**Pull request by jcouv**
Porting test involving conversion to base with tuple names differences. Tuple names should not matter when detecting whether a type is a base of another.

@dotnet/roslyn-compiler for review of RC fix.

Relates to https://github.com/dotnet/roslyn/pull/12904 (C# change that introduced this test)
This is part of a broader list of issues to check in VB: https://github.com/dotnet/roslyn/issues/13938

### CODE REVIEW

**Comment by VSadov** [comment]
LGTM

**Comment by jcouv** [comment]
@MattGertz Please approve for RC. This scenario should report an error but instead it compiles.

File: src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb [hunk 1]
```diff
@@ -717,7 +717,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     current.OriginalDefinition.AddUseSiteDiagnostics(useSiteDiagnostics)
                 End If
 
-                If current = superType Then
+                If current.IsSameType(superType, TypeCompareKind.IgnoreTupleNames) Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb]: >IsSameType [](start = 27, length = 10)

It looks like IsDerivedFrom is using IsSameTypeIgnoringAll. It feels like they should be consistent. Perhaps should use IsSameTypeIgnoringAll here as well.

Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb]: I aligned the implementations and added a test. 
Just to confirm, we want a type to derive from another even if the modifiers are different?

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb]: Yes, I think this is what we want. Modifiers are not taken into account by the language. They are interesting only if we are comparing member "signatures" using run-time semantics.

**Comment by jcouv** [comment]
@dotnet-bot test windows_debug_unit32_prtest  please
// error MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.

**Comment by jcouv** [comment]
@dotnet-bot test windows_debug_unit32_prtest please

**Comment by AlekseyTs** [comment]
LGTM

**Comment by jcouv** [comment]
Thanks! Merged

**Pull request merged**